nautobot/docs/rest-api/filtering.md

```diff
@@ -52,8 +52,16 @@ To filter results by a custom field value, prepend `cf_` to the custom field nam
 GET /api/dcim/sites/?cf_foo=123
 ```
 
+Multi-value filtering is available for all field types and lookup expressions. When multiple values are supplied, an `or` lookup will be performed by the query. For example, the following query will return only sites where a custom field named `foo` is equal to 123 _or_ 456:
```

Comment:
I thought that in #2126, for now we have only single-value support for the pre-existing default/exact lookup filter on custom fields? That multi-value was only for the newly implemented lookup-expression cases below?

Reply:
The default filter is still single value for backward compatibility. So for an `integer` custom field you would have these filters.

- Single value filter
  - `cf_cf1` (exact)
- Multi value filters
  - `cf_cf1_n` (exact negated)
  - `cf_cf1_lte` (less than or equal)
  - `cf_cf1_lt` (less than)
  - `cf_cf1_gte` (greater than or equal)
  - `cf_cf1_gt` (greater than)

One more thing to point out is not all custom field types got the extended lookup expression filters. We added extended filters for `text`, `url`, `select`, `integer` and `date` custom field types but not `boolean`, `multi-select` or `json`. We could add support for `multi-select` and `json` but those are going to require additional logic and were out of scope for the requested feature. I would just point out that the extended filters are supported in `text`, `url`, `select`, `integer` and `date` fields. The filters added are documented in the summary of #2126

Reply:
So I feel like the example in this PR that's doing an exact match is still incorrect:

`GET /api/dcim/sites/?cf_foo=123&cf_foo=456`

because this is an exact match, it will *not* do a multi-value search - instead the last specified value will "win" as the single value to filter on, and the previous values will be ignored, i.e. this will only filter for `cf_foo=456`.

https://next.demo.nautobot.com/api/dcim/sites/?cf_site_type=POP&cf_site_type=NON_POP demonstrates this - it only shows sites whose `site_type` CF is set to `NON_POP`, not also those set to `POP`.